compiler/src/value/integer/integer.rs

```diff
@@ -86,32 +86,19 @@ impl Integer {
         match_integer!(integer => integer.to_bits_le())
     }
 
-    // pub fn get_bits_typed(&self) -> (Vec<Boolean>, IntegerType) {
-    //     let integer = self;
-    //     (match_integer!(integer => integer.to_bits_le()), self.get_type())
-    // }
-
-    // pub fn from_bits_typed(type_: &IntegerType, bits: &[Boolean]) -> Integer {
-    //     match type_ {
-    //         IntegerType::U8 => Integer::U8(UInt8::from_bits_le(bits)),
-    //         IntegerType::U16 => Integer::U16(UInt16::from_bits_le(bits)),
-    //         IntegerType::U32 => Integer::U32(UInt32::from_bits_le(bits)),
-    //         IntegerType::U64 => Integer::U64(UInt64::from_bits_le(bits)),
-    //         IntegerType::U128 => Integer::U128(UInt128::from_bits_le(bits)),
-    //         IntegerType::I8 => Integer::I8(Int8::from_bits_le(bits)),
-    //         IntegerType::I16 => Integer::I16(Int16::from_bits_le(bits)),
-    //         IntegerType::I32 => Integer::I32(Int32::from_bits_le(bits)),
-    //         IntegerType::I64 => Integer::I64(Int64::from_bits_le(bits)),
-    //         IntegerType::I128 => Integer::I128(Int128::from_bits_le(bits)),
-    //     }
-    // }
+    pub fn is_allocated(&self) -> bool {
+        self.get_bits().into_iter().any(|b| !matches!(b, Boolean::Constant(_)))
```

Comment:
This logic says that if any bit of the integer is constant then we treat the whole array is constant.

We should instead be checking that if any bit of the integer is allocated then we treat the whole array as allocated.

```suggestion
        self.get_bits().into_iter().any(|b| matches!(b, Boolean::Is(_) | Boolean::Not(_)))
```